lib/model/rwfolder.go

```diff
@@ -342,14 +339,12 @@ func (f *sendReceiveFolder) pullerIteration(ignores *ignore.Matcher, ignoresChan
 		}()
 	}
 
-	for i := 0; i < numPullers; i++ {
-		pullWg.Add(1)
-		go func() {
-			// pullerRoutine finishes when pullChan is closed
-			f.pullerRoutine(pullChan, finisherChan, requestLimiter)
-			pullWg.Done()
-		}()
-	}
+	pullWg.Add(1)
+	go func() {
+		// pullerRoutine finishes when pullChan is closed
+		f.pullerRoutine(pullChan, finisherChan)
```

Comment:
I'd just pass `pullWg` to pullerroutine instead of creating a new waitgroup inside it and having this "artificial" wait-group for a single puller routine.

Reply:
I disagree. It looks redundant, but doing it this way retains the behavior that the function call is synchronous, like all the other ones around it, which is easier to reason about.

Reply:
I don't understand the part about synchronous behaviour: As it is inside a `go func() {...}()` construct I'd think it's asynchronous anyway and there wouldn't be any different behaviour with `go f.pullerRoutine(pullChan, finisherChan, pullWg)`?

Reply:
Maybe my wording was incorrect. What I mean is that I think it's cleaner that, when you launch a goroutine, it's clear what will happen when it exits and how we know that it's exited. Passing a waitgroup and expecting the function to decrement it when it exits is a bit unusual - and we're giving the function stuff to do that is not part of its main concern. There's no need to make the function specific to being called in a goroutine, it can just do it's thing and the caller gets to manage state.